tools/depmod.c

```diff
@@ -2289,10 +2288,9 @@ static int output_symbols(struct depmod *depmod, FILE *out)
 static int output_symbols_bin(struct depmod *depmod, FILE *out)
 {
 	struct index_node *idx;
-	char aliasbuf[1024] = "symbol:";
-	_cleanup_(scratchbuf_release) struct scratchbuf salias =
-		SCRATCHBUF_INITIALIZER(aliasbuf);
-	const size_t baselen = sizeof("symbol:") - 1;
+	_cleanup_strbuf_ struct strbuf salias;
+	const char *base = "symbol:";
+	const size_t baselen = strlen(base);
```

Comment:
The project commonly uses `sizeof("string-literal") - 1` instead of `strlen("string-literal")`. I am guessing that's because older compilers could not deduce/optimise the latter to a compile-time constant.

Is that right, or am I missing something obvious?

Reply:
that's right